src/main/fc/fc_msp.c

```diff
@@ -286,15 +286,19 @@ void initActiveBoxIds(void)
         activeBoxIds[activeBoxIdCount++] = BOXHORIZON;
     }
 
+#ifdef BARO
     if (sensors(SENSOR_BARO)) {
         activeBoxIds[activeBoxIdCount++] = BOXBARO;
     }
+#endif
 
+#ifdef MAG
```

Comment:
is it better to cut `activeBoxIds[activeBoxIdCount++] = BOXHEADFREE;` and then using this 
```
    if (sensors(SENSOR_ACC)) {
         activeBoxIds[activeBoxIdCount++] = BOXANGLE;
          activeBoxIds[activeBoxIdCount++] = BOXHORIZON;
         activeBoxIds[activeBoxIdCount++] = BOXHEADFREE;
      }
```